examples/react-example/App.tsx

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable */
+import React from 'react'
+import Uppy from '@uppy/core'
+import Tus from '@uppy/tus'
+import Webcam from '@uppy/webcam'
+import RemoteSources from '@uppy/remote-sources'
+import { Dashboard, useUppyState } from '@uppy/react'
+
+import '@uppy/core/dist/style.css'
+import '@uppy/dashboard/dist/style.css'
+import '@uppy/drag-drop/dist/style.css'
+import '@uppy/file-input/dist/style.css'
+import '@uppy/progress-bar/dist/style.css'
+
+const metaFields = [
+  { id: 'license', name: 'License', placeholder: 'specify license' },
+]
+
+function createUppy() {
+  return new Uppy({ restrictions: { requiredMetaFields: ['license'] } })
+    .use(Tus, { endpoint: 'https://tusd.tusdemo.net/files/' })
+    .use(Webcam)
+    .use(RemoteSources, {
+      companionUrl: 'https://companion.uppy.io',
+    })
+}
+
+export default function App() {
+  // IMPORTANT: passing an initaliser function to useState
+  // to prevent creating a new Uppy instance on every render.
+  // useMemo is a performance hint, not a guarantee.
+  const [uppy] = React.useState(createUppy)
+  // You can access state reactively with useUppyState
+  const fileCount = useUppyState(
+    uppy,
+    (state) => Object.keys(state.files).length,
+  )
+  const totalProgress = useUppyState(uppy, (state) => state.totalProgress)
+  // Also possible to get the state of all plugins.
+  const plugins = useUppyState(uppy, (state) => state.plugins)
```

Comment:
maybe this could be simplified as
```suggestion
  const { fileCount, totalProgress, plugins } = useUppyState(
    uppy,
    (state) => ({
      fileCount: Object.keys(state.files).length,
      totalProgress: state.totalProgress,
      plugins: state.plugins,
      // ... anything else you might need
    }),
  )
```

Reply:
This is a bit of an anti-pattern in the selector/lenses pattern. Selector should have a single purpose and the selector determines how often the hook needs to re-render based on the subset of state it returns, which is very big and deep in your case.

Reply:
maybe we should add a comment about that